src/stores/notifications/RoomNotificationState.ts

```diff
@@ -81,7 +81,12 @@ export class RoomNotificationState extends EventEmitter implements IDestroyable,
     private updateNotificationState() {
         const before = {count: this.count, symbol: this.symbol, color: this.color};
 
-        if (this.roomIsInvite) {
+        if (RoomNotifs.getRoomNotifsState(this.room.roomId) === RoomNotifs.MUTE) {
```

Comment:
if you are calling this then you would also need to watch for account-data to spot changes to push rules which would cause this to change

Reply:
I thought about that, but I don't think we need to. We have no way of telling which room actually changed in the push rules and so would be mounting constantly. Further, users wouldn't be changing things from muted to unmuted without the user consciously being aware of the room, and so either the next message or them visiting it will update the badge states. We do not need to handle unmuted to muted because of the global remount.

Reply:
hmm, though we do dedupe here so maybe it's fine...

Reply:
> Further, users wouldn't be changing things from muted to unmuted without the user consciously being aware of the room, and so either the next message or them visiting it will update the badge states.

Lets say Room X has a badge count, I mute it, the badge count would remain but then the mute symbol would also show because of our re-render everything which this does not cover. Now we have to render both the mute symbol and a badge count which will show in the wrong order because we are lazy :D

Reply:
One of the 75 fires of the account data event probably make that go away. I've added it below in any case.